esmvaltool/diag_scripts/cos22esd/climate_change_hotspot.py

```diff
@@ -0,0 +1,225 @@
+            save_data(basename, provenance_record, self.cfg, period_anomaly)
+
+    def compute_rolling_means(self, cube):
+        """Compute the 10yr rolling mean anomaly timeseries with respect
+        to the baseline period for the input cube.
+        """
+        area_mean = area_statistics(cube, 'mean')
+        anomaly = anomalies(area_mean, "full",
+                            reference=self.anomaly_reference)
+        if cube.var_name == "pr":
+            anomaly = self.relative_change(anomaly, area_mean)
+        timeseries_cube = anomaly.rolling_window(
+            "time", iris.analysis.MEAN, 2)  # 10 10 10 10 10 10 10 10 10
+        return timeseries_cube
+
+    def relative_change(self, anomaly, cube):
+        baseline_cube = extract_time(cube,
+                                     int(self.cfg["baseline_period"][0]),
+                                     1, 1,
+                                     int(self.cfg["baseline_period"][1]),
+                                     12, 31)
+        baseline_cube = climate_statistics(baseline_cube)
+        anomaly /= baseline_cube
+        return anomaly * 100
+
+    def get_hotspot_provenance_record(self, attributes, period,
+                                      season, ancestor_files):
+        """Create a provenance record describing the
+        hotspot fields diagnostic data."""
+        baseline = self.cfg["baseline_period"]
+        caption = (f"{attributes['project']} {season.upper()} "
+                   f"{attributes['long_name']} differences between global "
+                   f"and {self.cfg['region_name']} anomalies of period "
+                   f"{period}, against the baseline period "
+                   f"{baseline[0]}-{baseline[1]}, according to.")
+
+        record = {
+            'caption': caption,
+            'statistics': ['mean', 'anomaly', 'diff'],  # ??
+            'domains': ['global', 'reg'],
+            'authors': [
+                'cos_josep',
+            ],
+            'references': [
+                'cos22esd',
+            ],
+            'ancestors': ancestor_files,
+        }
+        return record
+
+    def get_rolling_mean_provenance_record(self, attributes, season,
+                                           region, ancestor_files):
+        """Create a provenance record describing the
+        rolling mean diagnostic data."""
+        baseline = self.cfg["baseline_period"]
+        caption = (f"{attributes['project']} {season.upper()} 10 year "
+                   f"{attributes['long_name']} rolling mean between "
+                   f"{attributes['start_year']} and {attributes['end_year']}, "
+                   f"against the baseline period {baseline[0]}-{baseline[1]}, "
+                   f"according to the {region} mean")
+
+        if region == self.cfg["region_name"]:
+            domain = "reg"
+        elif region == "global":
+            domain = "global"
+        else:
+            domain = "reg"
```

Comment:
Maybe this can  be simplified

```suggestion
        domain = "reg"
        if region == "global"
            domain = "global" 
```